ml_genn/model.py

```diff
@@ -290,108 +303,229 @@ def convert_tf_model(tf_model, converter=Simple(),
+            mlg_model_inputs.append(mlg_layer)
+
+        # while there are still layers to traverse
+        while new_tf_layers:
+            new_tf_layer = new_tf_layers.pop()
+            traversed_tf_layers.add(new_tf_layer)
+
+            # get next TF layer to configure
+            for tf_layer in tf_out_layers_all[new_tf_layer]:
+                tf_in_layers = set(tf_in_layers_all[tf_layer])
+                tf_out_layers = set(tf_out_layers_all[tf_layer])
+
+                # skip if we still need to configure inbound layers
+                if not traversed_tf_layers.issuperset(tf_in_layers):
+                    continue
+
+                # add this layer to new layers list
+                new_tf_layers.add(tf_layer)
+
+                # traverse ignored layers to find more inputs
+                new_tf_in_layers = tf_in_layers.copy()
+                while new_tf_in_layers:
+                    tf_in_layer = new_tf_in_layers.pop()
+                    if isinstance(tf_in_layer, ignored_tf_layers):
+                        new_tf_in_layers.update(tf_in_layers_all[tf_in_layer])
+                        tf_in_layers.discard(tf_in_layer)
+                    else:
+                        tf_in_layers.add(tf_in_layer)
+
+                # configure layer
+                print('configuring {} layer <{}>'.format(
+                    tf_layer.__class__.__name__, tf_layer.name))
+
+                # === Dense Layers ===
+                if isinstance(tf_layer, tf.keras.layers.Dense):
+                    name = tf_layer.name
+                    weights = []
+                    sources = []
+                    synapses = []
+
+                    # create layer
+                    mlg_layer = Layer(name=name, neurons=converter.create_neurons(
+                        tf_layer, pre_compile_output))
+
+                    # create synapses
+                    for tf_in_layer in tf_in_layers:
+                        weights.append(tf_layer.get_weights()[0])
+                        sources.append(mlg_layer_lookup[tf_in_layer])
+
+                        if isinstance(tf_in_layer, tf.keras.layers.AveragePooling2D):
+                            synapses.append(AvePool2DDenseSynapses(
+                                units=tf_layer.units,
+                                pool_size=tf_in_layer.pool_size,
+                                pool_strides=tf_in_layer.strides,
+                                pool_padding=tf_in_layer.padding,
+                                connectivity_type=connectivity_type))
+
+                        else:
+                            synapses.append(DenseSynapses(units=tf_layer.units))
+
+                    # connect layer and set weights
+                    mlg_layer.connect(sources, synapses)
+                    mlg_layer.set_weights(weights)
+
+                    mlg_layer_lookup[tf_layer] = mlg_layer
+                    if len(tf_out_layers) == 0:
```

Comment:
Why does this make this layer an output?

Reply:
If no layer outputs, then it is probably an output layer. I could change it to accept a list of the tf_layers you want as outputs, but this seemed a reasonable default.

Reply:
ahh ok - that makes sense - a comment to explain might be nice. It's really annoying that you can output tensors at the TF level with ``tf_model.outputs`` but, as far as I can see, there's no way to go from that to the keras level layer object.

Reply:
There's a lot of trivial graph traversal things you probably ought to be able to do, but can't without ugly hacking, in Keras... :(